File: eva-core/src/main/java/io/nerv/core/mvc/vo/PageVo.java
package io.nerv.core.mvc.vo;

import lombok.Builder;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.Collections;
import java.util.List;

/**
 * 通用分页对象
 * @param <T>
 */
@Data
@Accessors(chain = true)
public class PageVo<T> {
    /**
     * 查询数据列表
     */
    private List<T> records = Collections.emptyList();

    /**
     * 总页数
     */
    private long total = 0;
    /**
     * 每页显示条数，默认 10
     */
    private long size = 10;

    /**
     * 当前页
     */
    private long current = 1;
}

File: eva-core/src/main/kotlin/io/nerv/core/mvc/vo/PageVo.kt
package io.nerv.core.mvc.vo

/**
 * 通用分页对象
 * @param <T>
</T> */
class PageVo<T> {
    /**
     * 查询数据列表
     */
    val records: List<T> = emptyList()

    /**
     * 总页数
     */
    val total: Long = 0

    /**
     * 每页显示条数，默认 10
     */
    val size: Long = 10

    /**
     * 当前页
     */
    val current: Long = 1
}